File: LC_JS-SOLUTIONS/977. Squares of a Sorted Array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function (nums) {
  let arr = [];
  let l = 0;
  let r = nums.length - 1;

  while (l <= r) {
    if (nums[l] * nums[l] < nums[r] * nums[r]) {
      arr.unshift(nums[r] * nums[r]);
      r -= 1;
    } else {
      arr.unshift(nums[l] * nums[l]);
      l += 1;
    }
  }
  return arr;
};

File: LC-Python-Solution/977. Squares of a Sorted Array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:

        res = [0]*len(nums)
        l = 0
        r = len(nums)-1
        i = len(nums)-1
        
        while l<=r:
            if nums[l]*nums[l]<nums[r]*nums[r]:
                res[i] = nums[r]*nums[r]
                r-=1
                i-=1
            
            else:
                res[i] = nums[l]*nums[l]
                l+=1
                i-=1
        
        return res
        








        